src/util/queue.h

```diff
@@ -0,0 +1,41 @@
+
+#include <pthread.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "../redisai_memory.h"
+#include "redismodule.h"
+
+#ifndef __QUEUE_H
+#define __QUEUE_H
+
+typedef struct queueItem {
+  struct queueItem *next;
+  struct queueItem *prev;
+  void *value;
+} queueItem;
+
+typedef struct queue {
+  queueItem *front;
+  queueItem *back;
+  void (*free)(void *ptr);
+  unsigned long len;
+} queue;
+
+queue *queueCreate(void);
+void queuePush(queue *queue, void *value);
+queueItem *queuePop(queue *queue);
+queueItem *queueFront(queue *queue);
+queueItem *queueNext(queueItem *item);
+queueItem *queueEvict(queue *queue, queueItem *item);
+long long queueLength(queue *queue);
+void queueRelease(queue *queue);
+
+typedef struct RunQueueInfo {
```

Comment:
I think `RunQueueInfo` doesn't belong to `util`

Reply:
We can keep it in redisai.c for the time being, but later on we can factor it out in a runqueue.c (not in util) along with code related to aynchronous processing.